### PR TITLE
Environment parameters module port

### PR DESCRIPTION
**Pull request by EM-Creations**
**When merged this pull request will:**
- Port the environment parameters module to the new format.

### CODE REVIEW

File: modules/env_parameters/functions/ENV/fn_ENV_preInit.sqf [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#include "script_component.hpp"
+#include "..\..\settings.sqf"
+
+if (isMultiplayer) then {
+	private ["_WeatherParam", "_FogParam", "_overcast", "_wind", "_windDir", "_fog"];
```
Comment (PiZZAD0X) [inline modules/env_parameters/functions/ENV/fn_ENV_preInit.sqf:5]: private <Array> method is much slower than doing private _var method, it's a holdover from before private _var was introduced.

File: modules/env_parameters/functions/ENV/fn_ENV_preInit.sqf [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#include "script_component.hpp"
+#include "..\..\settings.sqf"
+
+if (isMultiplayer) then {
```
Comment (PiZZAD0X) [inline modules/env_parameters/functions/ENV/fn_ENV_preInit.sqf:4]: if you just want to exit the script on a condition use exitwith

Comment (EM-Creations) [reply modules/env_parameters/functions/ENV/fn_ENV_preInit.sqf]: Fixed.

File: modules/env_parameters/functions/ENV/fn_ENV_preInit.sqf (hunk 1 above)
Comment (EM-Creations) [reply modules/env_parameters/functions/ENV/fn_ENV_preInit.sqf]: Good to know! Fixed.

**Comment by EM-Creations** [comment]
> very nice idea with the timelimit addition, just needs the `private _var` changes and should be good to go

I can't take credit for it sadly, this originally came from an unmerged PR to the original Olsen Framework and I had incorporated it into mine.

**Review by PiZZAD0X — CHANGES_REQUESTED** [review]

**Review by PiZZAD0X — APPROVED** [review]

**Pull request merged**